### PR TITLE
live poll sockets

### PR DESCRIPTION
**Pull request by yassinmziya**
Sockets for live results

### CODE REVIEW

File: src/PollSocket.js [hunk 1]
```diff
@@ -137,6 +137,38 @@ export default class PollSocket {
...
+      }
+      if(question !== answer.question) {
+        console.log(`Question ${answer.question} is not the current question`);
+        return;
+      }
+
+      let nextState = {...this.current};
+      const prev = nextState.answers[answer.deviceId];
+      nextState.answers[answer.deviceId] = answer.data; // update/input user's response
+      if(!prev) {
```
Comment (meganle) [inline src/PollSocket.js]: should this be just `prev` since your checking if they previously responded?

File: src/PollSocket.js [hunk 2]
```diff
@@ -137,6 +137,38 @@ export default class PollSocket {
...
+        // has selected something before
+        nextState.results[prev] -= 1;
+      }
+
+      let cur_tally = nextState.results[answer.data];
+      if(!cur_tally) {
```
Comment (meganle) [inline src/PollSocket.js]: should this be just `cur_tally` since you increment if it already exists?

Comment (yassinmziya) [reply src/PollSocket.js]: !null ~ true. So if the option has been voted for, it's just incremented. but, in the case that the option hasn't been seen before, it is added to the result dictionary before anything happens..

File: src/PollSocket.js (hunk 1 above)
Comment (yassinmziya) [reply src/PollSocket.js]: Same here. !null ~ true i believe

File: src/PollSocket.js (hunk 2 above)
Comment (meganle) [reply src/PollSocket.js]: I think `!cur_tally` checks if `cur_tally` is falsey. [Check this out](https://stackoverflow.com/questions/5101948/javascript-checking-for-null-vs-undefined-and-difference-between-and/5101991)

Comment (yassinmziya) [reply src/PollSocket.js]: I used the safari console to verify. i think it's the appropriate check
screen shot 2018-02-11 at 15 52 04

**Pull request merged**